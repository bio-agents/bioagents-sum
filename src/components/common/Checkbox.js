import React from 'react'
import { Col, FormGroup } from 'react-bootstrap'

const Checkbox = ({ input, label }) => (
  <FormGroup>
    <Col smOffset={2} sm={10}>
      <div style={{ marginRight: 0 }} className='pretty p-icon p-smooth p-round p-bigger'>
        <input
          type='checkbox'
          checked={!!input.value}
          onChange={input.onChange}
        />
        <div className='state p-success'>
          <i className='icon fa fa-check' aria-hidden='true' />
          <label>{label}</label>
        </div>
      </div>
    </Col>
  </FormGroup>
)

export default Checkbox
