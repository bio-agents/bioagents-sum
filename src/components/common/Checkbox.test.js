import React from 'react'
import renderer from 'react-test-renderer'
import Checkbox from './Checkbox'

it('Renders Checkbox correctly', () => {
  const component = renderer.create(
    <Checkbox label='Hello, check me' input={{ value: true, onChange: () => !this.props.value }} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
