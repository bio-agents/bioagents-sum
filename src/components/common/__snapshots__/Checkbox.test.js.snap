exports[`test Renders Checkbox correctly 1`] = `
<div
  className="form-group">
  <div
    className="col-sm-10 col-sm-offset-2">
    <div
      className="pretty p-icon p-smooth p-round p-bigger"
      style={
        Object {
          "marginRight": 0,
        }
      }>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox" />
      <div
        className="state p-success">
        <i
          aria-hidden="true"
          className="icon fa fa-check" />
        <label>
          Hello, check me
        </label>
      </div>
    </div>
  </div>
</div>
`;
